collectd/collectd.go

```diff
@@ -107,7 +107,11 @@ func (s *Server) serve() {
 			continue
 		}
 		if n > 0 {
-			s.handleMessage(buffer[:n])
+			var messageBuffer = make([]byte, len(buffer))
+			copy(messageBuffer, buffer)
+
+			// Make packet processing in a separate goroutine so it prevents UDP packet loss
```

Comment:
I'm working with the person which opened the discussion in the mailing-list about this problem in parallel.  We worked together to make this work 0.8 and the results were that got less packet loss than before but he noticed a big increase in the program memory. 

We're still working on this issue and there are different ways to approach this (maybe using a pool of goroutines). We'll keep running tests and let u know our results